### ISSUE TITLE
install-app-deps successfully rebuilds node modules while node-gyp-rebuild fails

### ISSUE THREAD

**Issue by nikitakot** [electron-userland/electron-builder#7753]
* **Electron-Builder Version**:  24.6.3

* **Node Version**:  v18.16.1

* **Electron Version**: 25.2.0
* **Electron Type (current, beta, nightly)**: current

* **Target**: win32 x64

When running `npx electron-builder node-gyp-rebuild` node-gyp fails to rebuild native modules, parts of the error stack:
```shell
❯ npx electron-builder node-gyp-rebuild
  • executing @electron/rebuild  arch=1 version=25.2.0 appDir=C:\Users\nikitakot\code\app
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  napi.cpp
  LINK : fatal error LNK1181: cannot open input file 'C:\\Users\\nikitakot\\.electron-gyp\\25.2.0\\1\\node.lib'
...
Error: `C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
...

  ⨯ node-gyp failed to rebuild
```

When running `npx electron-builder install-app-deps` modules are rebuilt correctly and I have no issues using them in my electron application.

```shell
❯ npx electron-builder install-app-deps
  • electron-builder  version=24.4.0
  • executing @electron/rebuild  arch=x64 version=25.2.0 appDir=C:\Users\nikitakot\code\app
```

Which command should I use and what's the difference between them?

In the electron-builder config file I disable `nodeGypRebuild: false`, because I need to sign and apply additional version recourses to the module binaries, so I also need to use either `electron-builder node-gyp-rebuild` or `electron-builder install-app-deps` during the production build process.

**Comment by nikitakot** [comment]
@mmaietta thanks, the version bump helped, `npx electron-builder node-gyp-rebuild` now rebuilds modules as expected

the `npx electron-builder install-app-deps` though do not rebuild/build modules after the bump, so what is this command for? do i need to run it for the production build?

**Comment by mmaietta** [comment]
What about `./node_modules/.bin/electron-builder install-app-deps`?

**Comment by nikitakot** [comment]
@mmaietta 
```shell
❯ ./node_modules/.bin/electron-builder install-app-deps
  • electron-builder  version=24.6.4
```

**Comment by rotu** [comment]
Root cause https://github.com/npm/cli/issues/6662 is now fixed in [npm@10.1.0](https://github.com/npm/cli/releases/tag/v10.1.0)!
